documentation/Functions.md

```diff
@@ -1,3 +1,44 @@
-You can use the following function:
+## Functions 
+The following functions are already implemented:
 * installDevonIde
-* installCobiGen
\ No newline at end of file
+* installCobiGen
+* createProject
+* cobiGenJava
+
+***
+
+### installDevonIde
+#### parameter
+1. The tools you want to install within the devonfw ide: string array
+#### example
+installDevonfwIde(["java","mvn"])
+
+***
+
+### installCobiGen
+#### parameter
+* No parameters
+#### example
+installCobiGen()
+
+***
+
+### cobiGenJava
+#### parameter
+1. The path to the java file you want to generate code for: string
+2. The numbers that represent the templates that CobiGen uses to generate code: int array
+#### example
+cobiGenJava("path/to/java/file/MyEntity.java",[1,3,5,6,8])
+
+
+***
+
+### createProject
+#### parameter 
+1. the programming language that you will use for this project
+2. the project name
+
+#### example 
+createProject("java","cobigenexample")
```

Comment:
let's not focus on the programming language as a first parameter as there might be different architectures for one programming language bound to different project templates in near future.
So let's focus on a parameter like devon4j to indicate devon4j project template.

Reply:
Do you mean a differentiation between devon4j, devon4net, etc?
Should we create separate tasks for these?

Reply:
Yes and even within devon4j there might be more than one option to create a project.

I am not sure, possibly it's more simpler to have a task per stack like devon4j vs devon4ng.